server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -705,9 +712,8 @@ public Node(final Environment environment,
                     b.bind(SnapshotShardsService.class).toInstance(snapshotShardsService);
                     b.bind(RestoreService.class).toInstance(restoreService);
                     b.bind(Discovery.class).toInstance(discoveryModule.getDiscovery());
+                    b.bind(LogicalReplicationSettings.class).toInstance(replicationSettings);
                     {
-                        RecoverySettings recoverySettings = new RecoverySettings(settings,
-                                                                                 settingsModule.getClusterSettings());
                         processRecoverySettings(settingsModule.getClusterSettings(), recoverySettings);
```

Comment:
As `processRecoverySettings` belongs to the recovery settings, we should also move this up.

Reply:
After addressing comment below and getting rid of RecoverySettings dependency this line is back to original state as I